src/test/java/uk/gov/pay/api/it/ResourcesFilterITestBase.java

```diff
@@ -1,65 +1,104 @@
-    protected static final ZonedDateTime TIMESTAMP = DateTimeUtils.toUTCZonedDateTime("2016-01-01T12:00:00Z").get();
-    protected static final String CREATED_DATE = ISO_INSTANT_MILLISECOND_PRECISION.format(TIMESTAMP);
-    protected static final Map<String, String> PAYMENT_CREATED = new ChargeEventBuilder(CREATED, CREATED_DATE).build();
-    protected static final List<Map<String, String>> EVENTS = Collections.singletonList(PAYMENT_CREATED);
-    protected static final Address BILLING_ADDRESS = new Address("line1", "line2", "NR2 5 6EG", "city", "UK");
-    protected static final CardDetails CARD_DETAILS = new CardDetails("1234", "123456", "Mr. Payment", "12/19", BILLING_ADDRESS, "Visa");
-
-    protected static final String PAYLOAD = paymentPayload(AMOUNT, RETURN_URL, DESCRIPTION, REFERENCE);
-    protected ExecutorService executor = Executors.newFixedThreadPool(2);
+abstract public class ResourcesFilterITestBase {
+
+    static final String API_KEY = ApiKeyGenerator.apiKeyValueOf("TEST_BEARER_TOKEN", "qwer9yuhgf"); //Must use same secret set in test-config.xml's apiKeyHmacSecret
+    static final ZonedDateTime TIMESTAMP = DateTimeUtils.toUTCZonedDateTime("2016-01-01T12:00:00Z").get();
+    static final String PAYMENTS_PATH = "/v1/payments/";
+    static final int AMOUNT = 9999999;
+    static final String CHARGE_ID = "ch_ab2341da231434l";
+    static final PaymentState CREATED = new PaymentState("created", false, null, null);
+    static final String RETURN_URL = "https://somewhere.gov.uk/rainbow/1";
+    static final String REFERENCE = "Some reference";
+    static final String DESCRIPTION = "Some description";
+    static final String CREATED_DATE = ISO_INSTANT_MILLISECOND_PRECISION.format(TIMESTAMP);
+    static final List<Map<String, String>> EVENTS = List.of(new ChargeEventBuilder(CREATED, CREATED_DATE).build());
+    static final String GATEWAY_ACCOUNT_ID = "GATEWAY_ACCOUNT_ID";
+    static final String PAYLOAD = paymentPayload();
+    
+    private static final int CONNECTOR_PORT = findFreePort();
+    private static final int PUBLIC_AUTH_PORT = findFreePort();
+
+    private ExecutorService executor = Executors.newFixedThreadPool(2);
+    
+    @ClassRule
+    public static RedisDockerRule redisDockerRule;
+
+    static {
+        try {
+            redisDockerRule = new RedisDockerRule();
+        } catch (DockerException e) {
+            e.printStackTrace();
+        }
+    }
+    
+    @ClassRule
+    public static WireMockClassRule connectorMock = new WireMockClassRule(CONNECTOR_PORT);
+
+    @ClassRule
+    public static WireMockClassRule publicAuthMock = new WireMockClassRule(PUBLIC_AUTH_PORT);
+
+    @Rule
+    public DropwizardAppRule<PublicApiConfig> app = new DropwizardAppRule<>(
+            PublicApi.class, 
+            resourceFilePath("config/test-config.yaml"), 
+            config("connectorUrl", "http://localhost:" + CONNECTOR_PORT), 
+            config("connectorDDUrl", "http://localhost"), 
```

Comment:
Does this not need a port number and/or path?

Reply:
it doesn't because the classes that extend it don't use dd connector. i find setting unneeded dependencies to `http://localhost` useful because it gives us information that this test doesn't depend on dd connector.

Reply:
Ah, `http://unused-by-this-test.invalid` might be clearer but fine